### ISSUE TITLE
Appropriate configurations to get better rewind performance?

### ISSUE THREAD

**Issue by Sonia-7** [TASVideos/BizHawk#1829]
For processor I have a i7 7500U, and am running Bizhawk through a SSD with windows 10. For small savestates (NES, SMS, etc), I can enable rewind every 1 frame without the emulator lagging. For medium and large, if I leave them on default (every 4 frames and 60 respectively), I can keep a stable 60fps 100% of the time, but if I change those to every 1 frame (like on small states), the emulator starts stuttering a lot.

Before immediately dismissing it as a lack of processor horsepower, I'd like to know if there are any settings I can change in the Rewind & Savestate window to achieve a "every 1 frame" performance on medium/large savestates equal to that of small ones. There are plenty of options there such as Rewind Speed, Save State Definition, Compression Level, etc. So I'd like to know if changing those values around would help.

**Comment by zeromus** [comment]
yes.

**Comment by Sonia-7** [comment]
Judging from this answer, you only read the last bit of the second paragraph and replied to that alone. Let me rephrase things so it will no longer be a "yes or no" question:

What values should I put in them, and what checkboxes to leave enabled/disabled?

**Comment by zeromus** [comment]
Every possible value, in every combination, until you have identified the combination that performs the best

**Comment by Sonia-7** [comment]
All right. Thanks for this invaluable piece of information. It will surely come in hand.

No need to worry about this issue anymore, and sorry for any inconvenience. :)

**Comment by Sonia-7** [comment]
Actually, sorry for the double post but there's one last thing before (hopefully) putting this issue to rest.

So I did try many different combinations as you said, and eventually found what worked best. This:
https://i.imgur.com/LI7Kazr.png

From what I can understand, setting the Max buffer size to higher values allows rewinding for longer, while lower values can't go very far back in the past. A buffer of 64MB for example with those configurations only allows 1 minute of rewind. So my question is, what are the advantages of setting a lower buffer size?

**Comment by zeromus** [comment]
saving memory for higher buffer sizes elsewhere

**Comment by zeromus** [comment]
We could probably increase the default. It was picked when we probably all had 4GB of ram at most, and bizhawk was 32bit. I think we could probably set it to 2GB by default, no problem nowadays
